internal/backend/s3/s3.go

```diff
@@ -107,13 +107,13 @@ func getCredentials(cfg Config) (*credentials.Credentials, error) {
 	//    call to a pre-defined endpoint, only valid inside
 	//    configured ec2 instances)
 	creds := credentials.NewChainCredentials([]credentials.Provider{
+		&credentials.EnvAWS{},
```

Comment:
I've moved the `&credentials.EnvAWS{}` credentials back to the top of the list, as I'm pretty sure that it makes a difference. Only `EnvAWS` handles the `AWS_SESSION_TOKEN` environment variable.

Reply:
It could cause problems in my opinion.

You are correct that it handles AWS_SESSION_TOKEN, but static config should supersede all.

If you have env vars set and static config set, with your change the env vars will always take precedence which I feel is unexpected behavior.

So if static is first, if someone has passed in static creds it'll work as it does not require a session token.

Otherwise, if a session token is required then environment variables or an AWS profile is the only way to go.

Reply:
> It could cause problems in my opinion.

Well, before this PR `EnvAWS` was the first entry, thus keeping it at the position won't cause (new) problems.

In addition, the static config variables behave different than you expect: `cfg.KeyID` and `cfg.Secret` are filled from the environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`. An `AWS_SESSION_TOKEN` is always used together with a key_id and access_key, such that this combination would break if `credentials.Static` is moved before `EnvAWS`.

So let's just keep the order as it was before this PR.

Reply:
I see. That should probably not be the case, but I understand not changing it.

Just know that if the environment variables are set, it will always just use EnvAWS, the only time it'll use static is if the `cfg.KeyID` and `cfg.Secret` are explicitly set without environment variables.